tests/pjit_test.py

```diff
@@ -1295,18 +1295,88 @@ def f(x):
         pjit_f(jnp.array([1, 2, 3]))
 
   def test_pretty_print(self):
-    f = pjit(lambda x: x)
+    f = pjit(lambda x: x**2)
+    g = pjit(lambda x: f(x) + f(x))
     x = jnp.array([4.2], dtype=jnp.float32)
-    jaxpr = jax.make_jaxpr(f)(x)
+    jaxpr = jax.make_jaxpr(g)(x)
+    self.assertEqual(
+        jaxpr.pretty_print(),
+        textwrap.dedent("""
+            let <lambda> = { lambda ; a:f32[1]. let b:f32[1] = integer_pow[y=2] a in (b,) } in
+            { lambda ; c:f32[1]. let
+                d:f32[1] = pjit[
+                  name=<lambda>
+                  jaxpr={ lambda ; e:f32[1]. let
+                      f:f32[1] = pjit[name=<lambda> jaxpr=<lambda>] e
+                      g:f32[1] = pjit[name=<lambda> jaxpr=<lambda>] e
+                      h:f32[1] = add f g
+                    in (h,) }
+                ] c
+              in (d,) }
+        """).strip(),
+    )
+
+  def test_pretty_print_with_closure(self):
+    @pjit
+    def g(x, y):
+      @pjit
+      def f(x):
+        return x * y
+      return f(x) + f(y)
+
+    x = jnp.array([4.2], dtype=jnp.float32)
+    jaxpr = jax.make_jaxpr(g)(x, x)
     self.assertEqual(
         jaxpr.pretty_print(),
         textwrap.dedent("""
-            { lambda ; a:f32[1]. let
-                b:f32[1] = pjit[name=<lambda> jaxpr={ lambda ; c:f32[1]. let  in (c,) }] a
-              in (b,) }
+            let f = { lambda ; a:f32[1] b:f32[1]. let c:f32[1] = mul b a in (c,) } in
```

Comment:
The name probably should be unique wrt the variable names also (note `f` below) otherwise it will be confusing.

Reply:
Done.

The check I added is more conservative than necessary. We could make it more precise by tracking the parent jaxprs for each subjaxpr, but I thought it's an overkill.